### DISCUSSION TITLE
AGL in JSBSIM for Unreal goes to 0 when height over 11000 ft

### DISCUSSION THREAD

**Discussion by nmfarras** [JSBSim-Team/jsbsim#1105] [Q&A] [answered]
Hi I try to implement some feature that need AGL value as the reference.  I notice that when the value goes above around 11000 ft, the AGL automatically become 0. I turn the debugging feature but the trace line seems touching the ground. Any idea what's wrong with it?

Above 11000 ft
![image_2024-06-28_13-48-57](https://github.com/JSBSim-Team/jsbsim/assets/30061607/b01d657a-2f12-4c01-a960-d0dc1db030a8)

![image_2024-06-28_13-54-02](https://github.com/JSBSim-Team/jsbsim/assets/30061607/b8c8f031-8dcb-492c-abc5-0cd063615e87)

Below 11000 ft
![image_2024-06-28_13-55-29](https://github.com/JSBSim-Team/jsbsim/assets/30061607/3864b0fc-82c3-4f5a-bf37-62c3730611e9)

I try with the default model on UEReferenceApp too and the result more or less similar

Above:
![image_2024-06-28_14-28-14](https://github.com/JSBSim-Team/jsbsim/assets/30061607/5028a1a6-9376-4570-95e5-cb0e5f820e14)

Below 11000 ft:

![image_2024-06-28_14-28-38](https://github.com/JSBSim-Team/jsbsim/assets/30061607/6beac0a4-ec5b-47fb-bb89-da222d6f053a)

any insight? I may need the AGL can still be read around 25000 ft if possible

**Comment by gallonmate** [comment, marked as answer]
Yes there is bug with unit conversion. The code uses meters, instead of Unreal's centimeters.
The original code here on line137 says `FEET_TO_METER`
https://github.com/JSBSim-Team/jsbsim/blob/2a5985c2e190ca05f5ecd55b480eed40216dedb6/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModel/Private/JSBSimMovementComponent.cpp#L137

Open your JSBSimMovementComponent.cpp file and change that to `FEET_TO_CENTIMETER`, then compile. AGL will now work correctly at any altitude.

**Reply by nmfarras** [reply]
I never expect that was the case. Thank you for the quick response! Now it work without problem.

**Comment by seanmcleod** [comment]
@gallonmate will you submit a pull request with this change?

**Reply by seanmcleod** [reply]
@gallonmate would you prefer me to submit a pull request with your fix?

**Reply by gallonmate** [reply]
you can if you like, since I'm running behind on things. There's another tiny bug fix I was waiting to submit that's unrelated, but I was waiting for a chance to really sit down, build and test the latest UE5/JSBSim versions, and also re-visit some build issues there were noted by others. And I just haven't had the time. Feel free to submit. It could be a week or another month before I get around to it.

**Reply by seanmcleod** [reply]
Okay, fixed in commit - https://github.com/JSBSim-Team/jsbsim/commit/754a4aca640fb008d921853d009135bf99723610

**Reply by gallonmate** [reply]
@seanmcleod thanks!
